plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -65,10 +62,10 @@ public function data_regeneration_is_in_progress(): bool {
 	 * Initiate a table regeneration process.
 	 */
 	public function initiate_regeneration() {
+		require_once( Constants::get_constant( 'ABSPATH' ) . 'wp-admin/includes/upgrade.php' );
 		update_option( self::CUSTOM_ORDERS_TABLE_DATA_REGENERATION_IN_PROGRESS, 'yes' );
 		update_option( self::CUSTOM_ORDERS_TABLE_DATA_REGENERATION_DONE_COUNT, 0 );
-
-		// TODO: Create the tables as appropriate, schedule the table filling in batches with Action Scheduler.
+		dbDelta( $this->data_store->get_schema() );
```

Comment:
This is another case where a separate `DbUtils` class would be appropriate I think. The `require_once` and the `dbDelta` could be properly encapsulated in a dedicated method in such a class, and this would also help later with unit tests.